std/haxe/macro/Expr.hx

```diff
@@ -693,23 +714,43 @@ typedef Metadata = Array<MetadataEntry>;
 typedef Field = {
 	/**
 		The name of the field.
+		
+		Haxe field names start with an underscore `_`, a dollar `$`, a lower-case
+		character `a-z` or an upper-case character `A-Z`. After that, any combination 
+		and number of `_`, `A-Z`, `a-z` and `0-9` may follow.
 	**/
 	var name : String;
 
 	/**
 		The documentation of the field, if available. If the field has no
 		documentation, the value is `null`.
 	**/
-	@:optional var doc : Null<String>;
+	@:optional var doc : String;
 
 	/**
 		The access modifiers of the field. By default fields have private access.
+		
+		 * `Access.APublic` Public access modifier, grants access from anywhere.
+		 * `Access.APrivate` Private access modifier, grants access to class and its 
+		    sub-classes only.
+		 * `Access.AStatic` Static access modifier.
+		 * `Access.AOverride` Override access modifier.
+		 * `Access.ADynamic` Dynamic (re-)bindable access modifier.
+		 * `Access.AInline` Inline access modifier. Allows expressions to be directly 
+		    inserted inplace of calls to them.
```

Comment:
Also, why list all this here if we document the constructors on the type anyway?